stixpy/product/sources/science.py

```diff
@@ -516,14 +555,117 @@ def times(self):
         return self.data["time"]
 
     @property
-    def durtaion(self):
+    def duration(self):
         """
         An `astropy.units.Quantiy` array giving the duration or integration time
         """
         return self.data["timedel"]
 
+    def crop_by_value(
```

Comment:
```suggestion
    def crop_by_values(
```

Reply:
Would this function take high-level objects, e.g. `Time`, `SkyCord`, `SpectralCoord`?  Or just lower level `Quantity`?

Reply:
Low level objects plus a few strings for different pixel/detector combos e.g `cpd.crop_by_vales([Time(...), Time(...)], None, 'big', [4, 10]*u.keV)`